mage_ai/frontend/components/PipelineDetail/utils.ts

```diff
@@ -35,3 +36,17 @@ export function initializeContentAndMessages(blocks: BlockType[]) {
     messages: messagesInit,
   };
 }
+
+export function updateCollapsedBlocks(blocks: BlockType[], pipelineUUID: string, newPipelineUUID: string) {
+  blocks.forEach((b) => {
+    set(
+      `${newPipelineUUID}/${b.uuid}/codeCollapsed`,
```

Comment:
Instead of setting individual keys for every single block, can you set a single key for the collapsed blocks and a property in that object for each block?